src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -98,79 +149,242 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
+     * @param shards the number of shards. Shards split your data across multiple pods so you can fit more data into an index.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, Integer replicas,
                                       Integer shards) {
-        validatePodIndexParams(indexName, dimension, environment, podType, null, replicas, shards, null);
-
         return createPodsIndex(indexName, dimension, environment, podType, null, replicas, shards, null, null, null);
     }
 
-    // Minimal + replicas, shards + metadata
+    /**
+     * Overload for creating a new pod index with environment, podType, replicas, shards, and metadataConfig.
+     *
+     * <pre>
+     *     import io.pinecone.clients.Pinecone;
+     *     import org.openapitools.client.model.CreateIndexRequestSpecPodMetadataConfig;
+     *
+     *     Pinecone client = new Pinecone.Builder("your-api-key").build();
+     *
+     *     CreateIndexRequestSpecPodMetadataConfig metadataConfig = new CreateIndexRequestSpecPodMetadataConfig().fields(Arrays.asList("genre", "year"));
+     *     client.createPodsIndex("my-index", 1536, "us-east4-gcp", "p1.x2", "euclidean", 2, 2, metadataConfig);
+     * </pre>
+     *
+     * @param indexName the name of the index to be created.
+     * @param dimension the number of dimensions for the index.
+     * @param environment the cloud environment where you want the index to be hosted.
+     * @param podType the type of pod to use. A string with one of s1, p1, or p2 appended with a "." and one of x1, x2, x4, or x8.
+     * @param replicas the number of replicas. Replicas duplicate your index. They provide higher availability and throughput and can be scaled.
+     * @param shards the number of shards. Shards split your data across multiple pods so you can fit more data into an index.
+     * @return {@link IndexModel} representing the created serverless index.
+     * @throws PineconeException if the API encounters an error during index creation or if any of the arguments are invalid.
+     */
     public IndexModel createPodsIndex(String indexName, Integer dimension, String environment,
                                       String podType, Integer replicas,
                                       Integer shards, CreateIndexRequestSpecPodMetadataConfig metadataConfig) {
-        validatePodIndexParams(indexName, dimension, environment, podType, null, replicas, shards, null);
-
         return createPodsIndex(indexName, dimension, environment, podType, null, replicas, shards, null,
                 metadataConfig,
                 null);
     }
 
-    // Max
+    /**
+     * Creates a new pod index with the specified parameters.
```

Comment:
SUPER NIT but I'd make `pod` --> `pods`, since the method is "createPod`s`Index" :)